src/Umbraco.Web.UI.Client/src/packages/multi-url-picker/link-picker-modal/link-picker-modal.element.ts

```diff
@@ -140,68 +193,144 @@ export class UmbLinkPickerModalElement extends UmbModalBaseElement<UmbLinkPicker
-					${this.#renderLinkUrlInput()} ${this.#renderLinkTitleInput()} ${this.#renderLinkTargetInput()}
-					${this.#renderInternals()}
+					${this.#renderLinkType()} ${this.#renderLinkAnchorInput()} ${this.#renderLinkTitleInput()}
+					${this.#renderLinkTargetInput()}
 				</uui-box>
 				<div slot="actions">
 					<uui-button label=${this.localize.term('general_close')} @click=${this._rejectModal}></uui-button>
 					<uui-button
 						color="positive"
 						look="primary"
 						label=${this.localize.term('general_submit')}
-						?disabled=${!this.value.link.url && !this.value.link.queryString}
-						@click=${this._submitModal}></uui-button>
+						?disabled=${!this.value.link.type}
+						@click=${this.#onSubmit}></uui-button>
 				</div>
 			</umb-body-layout>
 		`;
 	}
 
-	#renderLinkUrlInput() {
+	#renderLinkType() {
 		return html`
-			<umb-property-layout orientation="vertical">
-				<div class="side-by-side" slot="editor">
-					<umb-property-layout
-						orientation="vertical"
-						label=${this.localize.term('defaultdialogs_link')}
-						style="padding:0;">
-						<uui-input
-							slot="editor"
-							placeholder=${this.localize.term('general_url')}
-							label=${this.localize.term('general_url')}
-							.value=${this.value.link.url ?? ''}
-							?disabled=${this.value.link.unique ? true : false}
-							@change=${this.#onLinkUrlInput}
-							${umbFocus()}>
-						</uui-input>
-					</umb-property-layout>
-					${when(
-						!this._config.hideAnchor,
-						() => html`
-							<umb-property-layout
-								orientation="vertical"
-								label=${this.localize.term('defaultdialogs_anchorLinkPicker')}
-								style="padding:0;">
-								<uui-input
-									slot="editor"
-									placeholder=${this.localize.term('placeholders_anchor')}
-									label=${this.localize.term('placeholders_anchor')}
-									@change=${this.#onLinkAnchorInput}
-									.value=${this.value.link.queryString ?? ''}></uui-input>
-							</umb-property-layout>
-						`,
-					)}
+			<umb-property-layout
+				orientation=${this.#propertyLayoutOrientation}
+				label=${this.localize.term('linkPicker_modalSource')}
+				mandatory>
+				<div slot="editor">
+					${this.#renderLinkTypeSelection()} ${this.#renderDocumentPicker()} ${this.#renderMediaPicker()}
+					${this.#renderLinkUrlInput()} ${this.#renderLinkUrlInputReadOnly()}
 				</div>
 			</umb-property-layout>
 		`;
 	}
 
+	#renderLinkTypeSelection() {
+		if (this.value.link.type) return nothing;
+		return html`
+			<uui-button-group>
+				<uui-button
+					look="placeholder"
+					label=${this.localize.term('general_document')}
+					@click=${this.#triggerDocumentPicker}></uui-button>
+				<uui-button
+					look="placeholder"
+					label=${this.localize.term('general_media')}
+					@click=${this.#triggerMediaPicker}></uui-button>
+				<uui-button
+					look="placeholder"
+					label=${this.localize.term('linkPicker_modalManual')}
+					@click=${this.#triggerExternalUrl}></uui-button>
```

Comment:
Do we need to specify `type="button"` on these to ensure they aren't considered as submit buttons e.g. by screen readers? (I cannot remember what the default is)

Reply:
The default for `<uui-button>` is [`type=button`](https://github.com/umbraco/Umbraco.UI/blob/v1.12.2/packages/uui-button/lib/uui-button.element.ts#L65).

Reply:
Hmm, peculiar. That is the opposite of the native `<button>` !